### PR TITLE
Ensure HSE description and version is the first log message

### PR DESCRIPTION
**Pull request by alexttx**

### CODE REVIEW

**Comment by alexttx** [comment]
It bothers me that this isn't the first log message emitted by HSE
```
[HSE] hse_platform_init: Heterogeneous-memory Storage Engine: version r2.2.0-dev-121-g22742e55, image kmt
```
This PR makes that the first log message.

File: lib/util/include/hse_util/base.h [hunk 1]
```diff
@@ -47,7 +47,7 @@
 
 /* clang-format off */
 
-#define HSE_UTIL_DESC   "Heterogeneous-memory Storage Engine Utilities"
+#define HSE_DESC      "Heterogeneous-memory Storage Engine"
```
Comment (tristan957) [inline lib/util/include/hse_util/base.h:50]: This is like the worst of both worlds in terms of alignment :)

Comment (tristan957) [reply lib/util/include/hse_util/base.h:50]: Can you just change it to HSE_NAME? That is the name of project after all

File: lib/binding/kvdb_interface.c [hunk 2]
```diff
@@ -113,6 +115,9 @@ hse_init(const char *const config, const size_t paramc, const char *const *const
         return err;
     }
 
+    /* First log message w/ HSE version - after deserializing global params */
+    log_info_sync("%s: version %s, image %s", HSE_DESC, HSE_VERSION_STRING, hse_progname);
```
Comment (tristan957) [inline lib/binding/kvdb_interface.c:119]: What does image refer to?

Comment (alexttx) [reply lib/binding/kvdb_interface.c:119]: program using libhse (eg, kmt).

**Review by tristan957 — APPROVED** [review]

**Comment by beckerg** [comment]
This has always bugged me as well...  I think the original desire was to emit a structured log message, but that wasn't possible until well into the initialization...

**Pull request merged**